src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/model/translator.kt

```diff
@@ -5,92 +5,96 @@ import uk.gov.justice.digital.hmpps.prisonersearch.services.canonicalPNCNumberLo
 import uk.gov.justice.digital.hmpps.prisonersearch.services.canonicalPNCNumberShort
 import uk.gov.justice.digital.hmpps.prisonersearch.services.dto.OffenderBooking
 
-fun <P : Prisoner> translate(prisoner: P, ob: OffenderBooking, incentiveLevel: IncentiveLevel?): P {
-  prisoner.prisonerNumber = ob.offenderNo
-  prisoner.bookNumber = ob.bookingNo
-  prisoner.bookingId = ob.bookingId?.toString()
-  prisoner.pncNumber = ob.identifiers?.firstOrNull { i -> i.type == "PNC" }?.value
-  prisoner.pncNumberCanonicalShort =
+fun PrisonerA(ob: OffenderBooking, incentiveLevel: IncentiveLevel?) =
+  PrisonerA().apply { this.translate(ob, incentiveLevel) }
+
+fun PrisonerB(ob: OffenderBooking, incentiveLevel: IncentiveLevel?) =
+  PrisonerB().apply { this.translate(ob, incentiveLevel) }
+
+fun Prisoner.translate(ob: OffenderBooking, incentiveLevel: IncentiveLevel?) {
```

Comment:
this is same, just converted to extension function for better readability from the caller's perspective

Reply:
Could maybe drop `this` on every line?